db/compaction_picker.cc

```diff
@@ -1373,6 +1356,39 @@ bool LevelCompactionBuilder::PickFileToCompact() {
   return start_level_inputs_.size() > 0;
 }
 
+bool LevelCompactionBuilder::PickL0ToBaseCompaction() {
```

Comment:
@ajkr I don't know whether people are relying on any such behavior or not, so I suggest we are careful. I think we can have it even narrow than what you said. If number of overlapping L1 file is 0 or 1, we stop expanding if more files are involved if we add an older file. Would that be OK to you?